### DISCUSSION TITLE
Change flow control window size

### DISCUSSION THREAD

**Discussion by RightSeven** [microsoft/msquic#2630] [Q&A] [answered]
Hi, I want to do some research on flow control like dynamical change window size. 

Is there any API can use? 
Should I just change the value of `QUIC_DEFAULT_STREAM_FC_WINDOW_SIZE` and `QUIC_DEFAULT_CONN_FLOW_CONTROL_WINDOW`?

Please tell me how to do.

**Comment by nibanks** [comment, marked as answer]
`QUIC_DEFAULT_STREAM_FC_WINDOW_SIZE` allows you to change the initial flow control window size for streams, but MsQuic owns the logic to dynamically grow that based on app receive speed (see [here](https://github.com/microsoft/msquic/blob/main/src/core/stream_recv.c#L644)).  `QUIC_DEFAULT_CONN_FLOW_CONTROL_WINDOW` is used as a fixed window for the connection flow control.

**Reply by bizzbyster** [reply]
A follow on transport question @nibanks. Is it possible to enable a non-default Congestion Control Algorithm (CCA) like Hybla or BBR? Thanks!

**Reply by nibanks** [reply]
It's possible to add additional CC algorithms, yes. But they have to be built-in to MsQuic. For instance, see this (draft) PR to add BBR support: https://github.com/microsoft/msquic/pull/2071.

**Reply by bizzbyster** [reply]
Interesting - ok I will check this out. As you probably guessed, we would want to enable something like BBR or Hybla (optimized for high latency) to a VpnHood deployment once it has QUIC between client and server as per our discussion here https://github.com/vpnhood/VpnHood/issues/94.

**Reply by nibanks** [reply]
Yeah, makes sense. Cubic still does a really good job (our current default), so I think it should be more than good enough to start with.
